Unusable for streaming on Windows 11

### mpv Information

```bash
mpv v0.41.0-dev-g41f6a6450 Copyright © 2000-2025 mpv/MPlayer/mplayer2 projects
 built on Dec 21 2025 19:30:47
libplacebo version: v7.358.0
FFmpeg version: f853d12
FFmpeg library versions:
   libavcodec      62.11.100
   libavdevice     62.1.100
   libavfilter     11.4.100
   libavformat     62.3.100
   libavutil       60.8.100
   libswresample   6.1.100
   libswscale      9.1.100
```

### Other Information

```text
Spare laptop for streaming tv:


Device name	HP
Processor	Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz (2.20 GHz)
Installed RAM	8.00 GB
Graphics card	Intel(R) HD Graphics 5500 (128 MB)
Storage	145 GB of 932 GB used
Device ID	38E0EE13-8494-4AF5-9066-9C51EF19FA42
Product ID	00330-80000-00000-AA882
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display


Windows version:


Edition	Windows 11 Pro
Version	25H2
Installed on	‎2026-‎03-‎05
OS build	26200.8457
Experience	Windows Feature Experience Pack 1000.26100.304.0
```

### Reproduction Steps

I'd guess the issue happened after Windows updated itself two weeks ago, as I have no other explanation and posting here

Initially I used scoop builds, then tried also v0.40, and then switched to several winget versions finally using:

```
Found MPV (Official CI build, MSVC) [mpv-player.mpv-CI.MSVC]
Version: 0.41.0
Publisher: MPV
Description: A free (as in freedom) media player for the command line. It supports a wide variety of media file formats, audio and video codecs, and subtitle types.
Homepage: https://github.com/mpv-player/mpv
License: LGPLv2.1
Tags:
  mpvplayer
  mpvci
  mpv.io
  mpvio
  media-player
  mediaplayer
  music-player
  musicplayer
  video-player
  videoplayer
Installer:
  Installer Type: portable (zip)
  Installer Url: https://github.com/mpv-player/mpv/releases/download/v0.41.0/mpv-v0.41.0-x86_64-pc-windows-msvc.zip
  Installer SHA256: 4e197f729f5071c6772f35fffd96e0f36e3e8a044bd9479b136bb09b7c6a80ff
  Release Date: 2025-12-21
  Offline Distribution Supported: true
```

It is same with all versions - when I run mpv taksbar is not accessible, nor Windows at all and I have hard time to actually close mpv
If I use VideoLan or any other player there is no issue

### Expected Behavior

Just play, without degrading

### Actual Behavior

As mentioned Windows UI is not accessible, I cannot click on taskbar not launch start menu while it shows after some period just enough to run task manager and terminate mpv. It doesn't use excessive CPU just it makes system not accessible

### Log File

> mpv .\tv.m3u
[mpvremote] Error: cannot open file: 'C:\Users\klo\AppData\Local\Temp/file-local-options.txt'
[mpvremote] true
[mpvremote] {"error_string":"","killed_by_us":false,"status":1,"stderr":"node:internal/modules/cjs/loader:1522\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\klo\\AppData\\Roaming\\mpv\\scripts\\mpvremote\\remoteServer.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1519:15)\r\n    at wrapResolveFilename (node:internal/modules/cjs/loader:1073:27)\r\n    at defaultResolveImplForCJSLoading (node:internal/modules/cjs/loader:1097:10)\r\n    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1124:12)\r\n    at Module._load (node:internal/modules/cjs/loader:1296:5)\r\n    at wrapModuleLoad (node:internal/modules/cjs/loader:255:19)\r\n    at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:154:5)\r\n    at node:internal/main/run_main_module:33:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v26.2.0\r\n"}
[mpvremote] ""
[mpvremote] Command subprocess: argument args has incompatible type.
[mpvremote] Mediastatus updated
Playing: http://192.168.100.6:9981/stream/channelid/525727695?ticket=C0BC9D64741ED8A24CD08B9C46B6F25794A21FF9
[mpvremote] Error: cannot open file: 'C:\Users\klo\AppData\Local\Temp/file-local-options.txt'
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
[ffmpeg/video] mpeg2video: Invalid frame dimensions 0x0.
 ● Video  --vid=1  (mpeg2video 720x576 25 fps)
 ● Audio  --aid=1  (mp2 2ch 48000 Hz 128 kbps)
AO: [wasapi] 48000Hz stereo 2ch float
VO: [gpu-next] 720x576 => 1024x576 yuv420p
[mpvremote] Mediastatus updated
AV: 00:00:14 / 00:00:15 (93%) A-V:  0.000 Cache: 0.6s/543KB
Exiting... (Quit)

### Sample Files

_No response_

### I carefully read all instruction and confirm that I did the following:

- [x] I tested and confirmed that the issue exists with the [latest release version](https://github.com/mpv-player/mpv/releases/latest) or newer.
- [x] I provided all required information including system and mpv version.
- [x] I produced the log file with the exact same set of files, parameters, and conditions used in "Reproduction Steps", with the addition of `--log-file=output.txt`.
- [x] I produced the log file while the behaviors described in "Actual Behavior" were actively observed.
- [x] I attached the full, untruncated log file.
- [x] I attached the backtrace in the case of a crash.

--- a/player/core.h
+++ b/player/core.h
@@ -269,6 +269,7 @@ typedef struct MPContext {
     char *term_osd_title;
     char *last_window_title;
     struct voctrl_playback_state vo_playback_state;
+    int64_t vo_playback_state_time;
 
     int add_osd_seek_info; // bitfield of enum mp_osd_seek_info
     double osd_visible; // for the osd bar only

--- a/player/misc.c
+++ b/player/misc.c
@@ -200,15 +200,25 @@ void update_vo_playback_state(struct MPContext *mpctx)
             .position = pos > 0 ? lrint(pos * UINT8_MAX) : 0,
         };
 
-        if (oldstate.taskbar_progress != newstate.taskbar_progress ||
-            oldstate.playing != newstate.playing ||
-            oldstate.paused != newstate.paused ||
-            oldstate.position != newstate.position)
-        {
+        bool state_changed = oldstate.taskbar_progress != newstate.taskbar_progress ||
+                             oldstate.playing != newstate.playing ||
+                             oldstate.paused != newstate.paused;
+        if (state_changed || oldstate.position != newstate.position) {
             // Don't update progress bar if it was and still is hidden
             if ((oldstate.playing && oldstate.taskbar_progress) ||
                 (newstate.playing && newstate.taskbar_progress))
             {
+                // Rate-limit progress-only updates. The win32 backend forwards
+                // these to the shell (explorer.exe), we don't want to spam it
+                // too much to avoid performance issues, when stealing cpu time
+                // from the shell thread. In normal playback, this doesn't update
+                // at all, but in live streams the duration can change, even on
+                // every frame.
+                int64_t now = mp_time_ns();
+                int64_t elapsed = now - mpctx->vo_playback_state_time;
+                if (!state_changed && elapsed < MP_TIME_MS_TO_NS(1000))
+                    return;
+                mpctx->vo_playback_state_time = now;
                 vo_control_async(mpctx->video_out,
                                  VOCTRL_UPDATE_PLAYBACK_STATE, &newstate);
             }

